### PR TITLE
fix(analytics): isolate public voting pages from PostHog

### PR DESCRIPTION
**Pull request by elboletaire**
This is claude-gpt-6-astra speaking on behalf of elboletaire.

## Findings

The existing event filter did not guarantee that PostHog was inactive on public voting pages:

- The initialization guard excluded process paths but missed the configured voting homepage.
- Same-document navigation could carry an already-running PostHog instance into a ballot. Dropping events does not stop SDK background activity.
- A pending SDK import could complete after navigation into a voting page.
- Encoded paths and configured language codes outside the original regex could bypass route detection.

Build and server configuration review confirmed that analytics and voting-homepage settings are supplied at runtime. No other repository was changed.

## Changes

- Disable Vike client routing for public process pages, their summaries, and the configured voting homepage, so they load in fresh documents.
- Replace the React Router ballot renderer with a document redirect to the same URL, preserving the language prefix and query string and handing rendering to Vike.
- Forward the pathname required for hydration on these server-routed pages.
- Pass runtime homepage and language configuration into the PostHog guard, safely decode path segments, and check the route both before and after importing the SDK.
- Prevent queued analytics helpers from calling the SDK when initialization was canceled.
- Keep event filtering as defense in depth, checking the actual browser path even when an event contains a stale URL.
- Add regression tests and document the privacy boundary.

A fresh document avoids relying on SDK teardown internals. The installed SDK shutdown method performs best-effort cleanup and flushes queued requests; it is not a guarantee of complete destruction. The destination voting document instead never initializes the app-owned PostHog instance.

## Verification

- `pnpm lint` — passed.
- `pnpm test` — 1,017 passed, 1 skipped; 173 test files passed, 1 skipped.
- `pnpm build` — passed.
- Ten controlled production-build browser checks with the real PostHog SDK passed: direct process and summary pages, root and localized voting homepages, encoded URLs, navigation from active PostHog pages, and programmatic React Router navigation into a ballot. Voting documents loaded no PostHog SDK resources or endpoint requests; transitions created new documents.
- Regression coverage includes route matching, deferred initialization, queued helper cancellation, provider configuration, document redirects, and hydration configuration.

## Scope and limitations

Live deployment settings and externally configured GTM tags were not inspected. This change isolates the application-owned PostHog integration; it does not change other analytics integrations or control scripts injected by external tags. The Docker-backed end-to-end suite was not run.

An additional process-card click check could not reach its link because the existing custom-domain homepage loader returns a different data shape than its organization view expects. That unrelated path is unchanged; the SPA process handoff was verified through its loader regression test and programmatic browser navigation.

### CODE REVIEW

File: src/utils/analytics.ts [hunk 1]
```diff
@@ -272,16 +316,46 @@ const loadPosthogModule = () => {
...
+// Shared body of every helper below: `posthogInitStarted` gates the synchronous
+// path and `posthogInitialized` the asynchronous one, so nothing reaches the SDK
+// when initialization never ran or was canceled by the voting-route guard.
+const withPosthog = (errorMessage: string, run: (posthog: Posthog) => void): void => {
+  if (!posthogInitStarted) return
+  if (!canUseBrowserAnalytics()) return
+
+  void loadPosthogModule()
+    .then(({ default: posthog }) => {
+      if (!posthogInitialized) return
```
[the remark this answers is not in the document]
Comment (elboletaire) [reply src/utils/analytics.ts:330]: But posthogInitialized is a global variable defined in the file. Even though this execution might not change anything, I don't think it's a bad practice to enforce that return in there, in case the value changed outside while executing the promise.

**Review by elboletaire — COMMENTED** [review]
works
